src/javascript/binary/websocket_pages/user/account/settings/iphistory/iphistory.ui.js

```diff
@@ -1,6 +1,38 @@
 var IPHistoryUI = (function(){
     "use strict";
 
+    function parse_ua(user_agent) {
+        // Table of UA-values (and precedences) from:
+        //  https://developer.mozilla.org/en-US/docs/Browser_detection_using_the_user_agent
+        var vn = '([0-9\.]+)';
+        var lookup = [
+            {name: 'Chromium',  regex: 'Chromium/' + vn},
+            {name: 'Chrome',    regex: 'Chrome/' + vn},
+            {name: 'Seamonkey', regex: 'Seamonkey/' + vn},
+            {name: 'Firefox',   regex: 'Firefox/' + vn},
+            {name: 'Opera',     regex: 'OPR/' + vn},
+            {name: 'Opera',     regex: 'Opera/' + vn},
+            {name: 'Safari',    regex: 'Version/' + vn},
+            {name: 'Internet Explorer', regex: ';MSIE '+ vn +';'},
+        ];
+        var len = lookup.length;
+        var name = 'Unknown';
+        var version = 'Unknown';
+        while (len--) {
```

Comment:
Yes, ``forEach`` is clearer but we need to exit early from the loop once we've "hit" a matching browser.